lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -850,6 +861,13 @@ public static function get_resolved_theme_uris( $theme_json ) {
 			return $resolved_theme_uris;
 		}
 
+		$theme_json_data               = $theme_json->get_raw_data();
+		$resolved_theme_uris_cache_key = md5( wp_json_encode( $theme_json_data ) );
```

Comment:
Not yet sure if this will make things better or worse 😄

Reply:
It'd be great if WordPress had minimum PHP 8 support for faster hashing: https://php.watch/articles/php-hash-benchmark

Reply:
`md4` outperforms `md5` for file integrity on many benchmarks I've seen but `hash(  'md4', wp_json_encode( $theme_json_data ) )` doesn't move the needle at all for me in performance tests.

Reply:
If we had a minimum PHP version of 8, we could avoid the hashing altogether and use a `WeakMap` where the theme_json data/object was the key.